src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -226,26 +228,20 @@ internal FixVersionOne CreateFix(Fix v2Fix)
             return fix;
         }
 
-        internal HashVersionOne CreateHash(Hash v2Hash)
+        internal IList<HashVersionOne> CreateHashVersionOneListFromV2Hashes(IDictionary<string, string> v2Hashes)
         {
-            HashVersionOne hash = null;
+            if (v2Hashes == null) { return null; }
+
+            var v1Hashes = new List<HashVersionOne>();
 
-            if (v2Hash != null)
+            foreach (string key in v2Hashes.Keys)
             {
-                AlgorithmKindVersionOne algorithm;
-                if (!Utilities.AlgorithmNameKindMap.TryGetValue(v2Hash.Algorithm, out algorithm))
-                {
-                    algorithm = AlgorithmKindVersionOne.Unknown;
-                }
+                Utilities.AlgorithmNameKindMap.TryGetValue(key, out AlgorithmKindVersionOne algorithm);
```

Comment:
>Utilities [](start = 16, length = 9)

Retain the logic that if this method returns `false`, `algorithm` should be `AlgorithmKindVersionOne.Unknown`.

Reply:
It does. Unknown is the first element of the enum, assigned a value of '0' (the value to which an enum variable is initialized).

---
In reply to: [226735147](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226735147) [](ancestors = 226735147)